storage/innobase/lock/lock0lock.cc

```diff
@@ -4498,64 +4523,108 @@ static void lock_rec_unlock(hash_cell_t &cell, lock_t *lock, ulint heap_no)
 }
 
 /** Release locks to unmodified records on a clustered index page.
-@param cell     lock_sys.rec_hash cell of lock
-@param lock     record lock
-@param offsets  storage for rec_get_offsets()
-@param heap     storage for rec_get_offsets()
-@param mtr      mini-transaction (will be started and committed) */
-static void lock_rec_unlock_unmodified(hash_cell_t &cell, lock_t *lock,
-                                       rec_offs *&offsets, mem_heap_t *&heap,
-                                       mtr_t &mtr)
-{
+@param block      the block containing locked records
+@param cell       lock_sys.rec_hash cell of lock
+@param lock       record lock
+@param offsets    storage for rec_get_offsets()
+@param heap       storage for rec_get_offsets()
+@param latch_type how the caller of the function latched lock_sys,
+                  GLOBAL or CELL
+@return true if the cell was latched successfully or if latch_type is GLOBAL,
+        false otherwise */
+static bool lock_rec_unlock_unmodified(buf_block_t *block, hash_cell_t *cell,
+                                       lock_t *lock, rec_offs *&offsets,
+                                       mem_heap_t *&heap,
+                                       lock_sys_latch_type latch_type)
```

Comment:
`latch_type` could also be a `template<bool global>`; the function looks rather small, so expanding it twice should not incur too much code bloat. It’s up to you; this should not be frequently executed or time-critical code, so replacing the run-time conditions on `latch_type` with machine code duplication might not matter much.